modules/nf-google/src/main/nextflow/cloud/google/pipelines/GooglePipelinesTaskHandler.groovy

```diff
@@ -109,11 +108,34 @@ class GooglePipelinesTaskHandler extends TaskHandler {
         if (!task.container) {
             throw new ProcessUnrecoverableException("No container image specified for process $task.name -- Either specify the container to use in the process definition or with 'process.container' value in your config")
         }
-        if (!instanceType) {
-            throw new ProcessUnrecoverableException("No instance type specified for process $task.name -- Please provide a 'cloud.instanceType' definition in your config")
+    }
+
+    //
+    // Use the process cpus and memory directives to create custom GCP instance
+    // If memory not specified, default to 1 GB per cpu.  An absence of the cpus directive defaults to 1 cpu.
+    // If the process machineType is defined, use that instead of cpus/memory (must be a predefined GCP machine type)
+    // The deprecated cloud.instanceType will be used if specified and machineType is not specified
+    String getProcessMachineType() {
+        String machineType = getProcessMachineType(executor.getSession().config.navigate("cloud.instanceType"), task.config.machineType, task.config.cpus, task.config.memory)
+
+        log.trace "[GPAPI] Task: $task.name - Instance Type: $machineType"
+
+        return machineType
+    }
+
+    String getProcessMachineType(String cloudInstanceType, String taskMachineType, int cpus, MemoryUnit memory) {
+
+        String machineType = taskMachineType ?: cloudInstanceType
+
```

Comment:
A message should be printed when using `cloud.instanceType` warning the user that the setting is deprecated should use `process.machineType` instead.

Reply:
You can use `log.warn1` for this, see for example [here](https://github.com/nextflow-io/nextflow/blob/b54dc046a0b0238bae6fbcc52d231bd77cf4b9f4/modules/nextflow/src/main/groovy/nextflow/processor/TaskProcessor.groovy#L554)` 

The `warn1` is an extension of `warn` that print only the very first occurrence of the same message.

Reply:
Added warn1 message